spec/forms/submission_form_spec.rb

```diff
@@ -0,0 +1,195 @@
+require 'rails_helper'
+
+RSpec.describe SubmissionForm do
+  let(:contact_method) { create :contact_method_email }
+  let(:location_type)  { create :location_type }
+  let(:service_area)   { create :service_area }
+
+  describe 'creating a new submission' do
+    let(:params) {{
+      form_name: 'Offer_form',
+      privacy_level_requested: 'anyone',
+      service_area: service_area.id,
+      listing_attributes: {
+        type: 'Offer',
+        tag_list: ['errands','groceries'],
+        description: 'on a quiet day i can hear her breathing',
+      },
+      location_attributes: {
+        id: nil,
+        city: 'Anakwashtank',
+        state: 'DC', # statehood now!
+        location_type: location_type.id
+      },
+      person_attributes: {
+        id: nil,
+        preferred_contact_method: contact_method.id,
+        email: 'we@together.coop',
+        name: 'Harriet Tubman',
+      },
+    }}
+
+    subject(:submission) { SubmissionForm.build params }
+
+
+    it 'builds a Submission instance' do
+      expect(submission).to be_a Submission
+    end
+
+    it 'populates submission fields' do
+      expect(submission.privacy_level_requested).to eq 'anyone'
+      expect(submission.form_name).to eq 'Offer_form'
+    end
+
+    it 'sets the specified service area' do
+      expect(submission.service_area).to eq service_area
+    end
+
+    it 'captures whole submission json'
+
+    describe 'has_one person' do
+      subject(:person) { submission.person }
+
+      it 'builds a Person instance' do
+        expect(person).to be_a Person
+      end
+
+      it 'populates person fields' do
+        expect(person.email).to eq 'we@together.coop'
+        expect(person.name).to eq 'Harriet Tubman'
+      end
+
+      it 'sets the specified contact method' do
+        expect(person.preferred_contact_method).to eq contact_method
+      end
+
+      describe 'has_one location' do
+        subject(:location) { person.location }
+
+        it 'builds a Location instance' do
+          expect(location).to be_a Location
+        end
+
+        it 'populates location fields' do
+          expect(location.city).to eq 'Anakwashtank'
+          expect(location.state).to eq 'DC'
+        end
+
+        it 'set the specified location_type' do
+          expect(location.location_type).to eq location_type
+        end
+      end
+    end
+
+    describe 'has_one listing' do # todo: change this to has_many
+      subject(:listing) { submission.listings.first }
+
+      it 'builds a Listing instance' do
+        expect(listing).to be_a Listing
+      end
+
+      it 'populates listing fields' do
+        expect(listing.type).to eq 'Offer'
+        expect(listing.description).to eq 'on a quiet day i can hear her breathing'
+      end
+
+      it 'populates categories on the listing' do
```

Comment:
should we also test it populates urgency level? (i think we aren't passing an urgency level at all rn, so, maybe not yet?)

Reply:
same for language. i wonder if we should set those like we do w contact_method_email? 

`preferred_locale_en`, and `urgency_level_within_a_week` ?

(i know this is more a refactor tho, so, sorry to muddy those waters

Reply:
I'm planning to add those as we go from one listing to listing-per-category.